packages/amplify-category-function/src/provider-utils/awscloudformation/utils/ensure-lambda-arn-outputs.ts

```diff
@@ -0,0 +1,30 @@
+import { AmplifyCategories, AmplifySupportedService, pathManager, readCFNTemplate, stateManager, writeCFNTemplate } from 'amplify-cli-core';
+import * as path from 'path';
+
+/**
+ * updates function cfn stack with lambda execution role arn parameter
+ */
+ export const ensureLambdaExecutionRoleOutputs = async (): Promise<void> => {
+    const amplifyMeta = stateManager.getMeta();
+    const functionNames = Object.keys(amplifyMeta?.[AmplifyCategories.FUNCTION] ?? []);
+    // filter lambda layer from lambdas in function
+    const lambdaFunctionNames = functionNames.filter(functionName => {
+      const functionObj = amplifyMeta?.[AmplifyCategories.FUNCTION]?.[functionName];
+      return functionObj.service === AmplifySupportedService.LAMBDA;
+    });
+    for (const functionName of lambdaFunctionNames) {
+      const templateSourceFilePath = path.join(pathManager.getBackendDirPath(), AmplifyCategories.FUNCTION, functionName, `${functionName}-cloudformation-template.json`);
+      const { cfnTemplate } = readCFNTemplate(templateSourceFilePath);
+      if (cfnTemplate.Outputs !== undefined && !cfnTemplate?.Outputs?.LambdaExecutionRoleArn) {
+        cfnTemplate.Outputs.LambdaExecutionRoleArn = {
+          Value: {
+            'Fn::GetAtt': [
+              'LambdaExecutionRole',
+              'Arn',
+            ],
+          },
+        };
+        await writeCFNTemplate(cfnTemplate, templateSourceFilePath);
```

Comment:
I'm concerned about writing to this file on push because it's not a build file and it's tracked by version control. Can we display a warning with a list of the templates that were modified automatically?

Reply:
This change is released with 10.6 . Should I add a warning now ?
This is done to incorporate hosting forcePush. For amplify-cli , this is done in prePush Handler

Reply:
I see, if this is already released in the pre-push handler, it's not a blocker. We can consider adding a warning in a future PR instead.